store/types/gas.go

```diff
@@ -63,21 +63,27 @@ func NewGasMeter(limit Gas) GasMeter {
 	}
 }
 
+// GasConsumed returns the gas consumed from the GasMeter.
 func (g *basicGasMeter) GasConsumed() Gas {
 	return g.consumed
 }
 
+// GasRemaining returns the gas left in the GasMeter.
 func (g *basicGasMeter) GasRemaining() Gas {
 	if g.IsPastLimit() {
 		return 0
 	}
 	return g.limit - g.consumed
 }
 
+// Limit returns the gas limit of the GasMeter.
 func (g *basicGasMeter) Limit() Gas {
 	return g.limit
 }
 
+// GasConsumedToLimit returns the gas limit if gas consumed is past the limit,
+// otherwise it returns the consumed gas.
+// NOTE: This behaviour is only called when recovering from panic when BlockGasMeter consumes gas past the limit.
```

Comment:
[https://github.com/cosmos/cosmos-sdk/pull/9665#discussion_r667481086](url) does this makes sense? @ValarDragon @alexanderbez

Reply:
```suggestion
// NOTE: This behaviour is only called when recovering from panic when
// BlockGasMeter consumes gas past the limit.
```